server/src/main/java/io/unitycatalog/server/persist/dao/ModelVersionInfoDAO.java

```diff
@@ -0,0 +1,101 @@
+package io.unitycatalog.server.persist.dao;
+
+import io.unitycatalog.server.model.ModelVersionInfo;
+import io.unitycatalog.server.model.ModelVersionStatus;
+import io.unitycatalog.server.persist.utils.FileUtils;
+import jakarta.persistence.*;
+import java.util.Date;
+import java.util.UUID;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+// Hibernate annotations
+@Entity
+@Table(
+    name = "uc_model_versions",
+    indexes = {
+      @Index(name = "idx_model_version", columnList = "registered_model_id,version"),
+    })
+// Lombok annotations
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@SuperBuilder
+public class ModelVersionInfoDAO {
+  @Id
+  @Column(name = "id")
+  private UUID id;
+
+  @Column(name = "registered_model_id")
+  private UUID registeredModelId;
```

Comment:
No foreign key reference to registered model ? Might help us in simplifying some cascaded fetch / delete operations. 
ref:
https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/dao/ColumnInfoDAO.java#L28
https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/dao/TableInfoDAO.java#L53

Reply:
So this isn't a strict FK relationship.   We *must* keep track of the registered model id on the model version in order to be able to list the model versions for a registered model.  DB UC also has this field defined.